hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/SerializableProxy.java

```diff
@@ -30,6 +35,8 @@ public final class SerializableProxy extends AbstractSerializableProxy {
 
 	private final CompositeType componentIdType;
 
+	private static volatile BytecodeProviderImpl defaultProvider;
```

Comment:
This is a bit tricky: if `defaultProvider` gets initialized (for wathever strange reason) during the before-bootstrap phase in a native image compilation, it will get stuck on the wrong reference.

Some options:

 A) Store it somewhere else in a non-static field (seems hard to find a sane place?)
 B) Not store it at all. Probably not good for performance but I'm not sure of how bad that would be, might not be too bad (?)
 C) Ensure this particular field gets reset by the GraalVM compiler; this could be done either via the annotation https://docs.oracle.com/en/graalvm/jdk/17/sdk/com/oracle/svm/core/annotate/RecomputeFieldValue.Kind.html ; we probably don't want the project to depend on such annotations but there should be an API one can use:
 - https://docs.oracle.com/en/graalvm/jdk/17/sdk/org/graalvm/nativeimage/hosted/Feature.BeforeAnalysisAccess.html#registerFieldValueTransformer-java.lang.reflect.Field-org.graalvm.nativeimage.hosted.FieldValueTransformer-  (let me know if you want to go down this path, in that case I can help..)

Reply:
Why would it be initialized, though? This method only gets called when trying to deserialize a `SerializableProxy` instance. Are there any reasons why this could happen during the "before-bootstrap" phase?

Apart from that, I'm not quite sure which of the options is the best one to follow to be honest, you can pick what you feel is the best way to go and I'll just update the PR with that.

Reply:
Yes I suppose you're right that the problem I raised is quite unlikely and difficult to trigger; it would require for some code to trigger serialization of some proxy before the classpath is changed - my concern stems from the possibility of `static` fields to be shared shared across multiple instances of Hibernate ORM, so if some SessionFactory(s) were to be initialized at a later stage, or perhaps rebooted, the cached field could witness a reuse of previous value - however I suppose in such a case we wouldn't also witness a change of the enumeration of service implementations available on the classpath, this would only be possible if the condition of multiple ORMs instances is combined with a native image compilation or some other runtime/container trick.

Feel free to ignore it for now - I admit it's not a practical problem as I don't think it can be triggered in practice in *today* in the frameworks we care about, I'll just prefer to avoid such possibilities as if it ever happens one day, perhaps because some of our assumptions about bootstrap are changed, it would be a very tricky one nail down.

In short, resetting the static field is the right thing to do in theory but I can take care of that later.